File: 2022- CS50 intro to programming (online course)/C programs/Earlier codes/readability.c
#include <cs50.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

float get_grade(void);
int letters;
int words;
int spaces;
int dot;
int sentences;
int grade;

int main(void)
{
    string input = get_string("Text: ");
    // Make sure random stuff before the text won't count
    spaces = 1;
    dot = 1;

    for (int i = 0; i < strlen(input); i++)
    {
        // Letter counter and word reset
        if ((input[i] >= 'A' && input[i] <= 'Z') || (input[i] >= 'a' && input[i] <= 'z'))
        {
            letters++;
            spaces = 0;
            dot = 0;
        }
        // Mark end of a word, count spaces
        if (input[i] == 32)
        {
            if (spaces == 0)
            {
                words++;
                spaces = 1;
            }
        }
        // Count periods and such as sentences
        if (input[i] == 33 || input[i] == 46 || input[i] == 63)
        {
            if (dot == 0)
            {
                sentences++;
                words++;
                spaces++;
                dot++;
            }
        }
    }
    grade = get_grade();

    if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade > 16)
    {
        printf("Grade 16+\n");
    }
    else
    {
        printf("Grade %i\n", grade);
    }
}

float get_grade(void)
{
    // Compile results from main
    float compile;
    compile = ((5.88 * letters - 29.6 * sentences) / words - 15.8);
    return round(compile);
}

File: 2022- CS50 intro to programming (online course)/Python programs/Early python/readability.py
# TODO
import cs50 as cs
# Program reads an input and calculates the readability grade


text = cs.get_string("Text: ")

letters = 0
words = 1
sentences = 0

# Count letters, spaces and periods
for i in range(len(text)):
    if text[i].isalpha():
        letters += 1

    if text[i] == ' ':
        words += 1

    if text[i] == "." or text[i] == "?" or text[i] == '!':
        sentences += 1

grade = round((5.88 * letters - 29.6 * sentences) / words - 15.8)

if grade < 1:
    print("Before Grade 1")
elif grade > 16:
    print("Grade 16+")
else:
    print("Grade ", end="")
    print(grade)